packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -165,12 +209,82 @@ export class PendingStateManager {
         this.pendingStates.push(pendingFlush);
     }
 
+    public rebaseAt(seqNum: number) {
+        const rebasePromises: Promise<void>[] = [];
+        // rebase initial ops at sequence number
+        while (!this.initialStates.isEmpty()) {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const nextState = this.initialStates.peekFront()!;
+            if (nextState.type === "message") {
+                if (nextState.referenceSequenceNumber > seqNum) {
+                    break; // nothing left to do at this sequence number
+                } else if (nextState.referenceSequenceNumber > 0 && nextState.referenceSequenceNumber < seqNum) {
```

Comment:
Should this be an assert instead of check? I.e. what are the other options? referenceSequenceNumber  can't be undefined, it can't be negative.